Atomic/discovery.py

```diff
@@ -21,17 +21,27 @@ def __init__(self, registry=None, repo=None, image=None, tag=None, digest=None,
         self.digest = digest
         self.orig_input = orig_input
         self._remote_inspect = None
-        self.fqdn = None
+        self._fqdn = None
 
         if self.debug:
             util.output_json(self.registries)
 
+    @property
+    def fqdn(self):
+        if not self._fqdn:
+            self._fqdn = self.assemble_fqdn(include_tag=True) if self.registry else self.find_image_on_registry()
+        return self._fqdn
+
+    @fqdn.setter
+    def fqdn(self, value):
+        self._fqdn = value
+
     def inspect(self):
+        if not self.fqdn:
```

Comment:
These two lines seem to be useless, since the first call will populate the self.fqdn?

I think you should just delete them and everything will be taken out.